flink-iotdb-connector/src/main/java/org/apache/iotdb/flink/IoTDBSink.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iotdb.flink;
+
+import com.google.common.collect.Lists;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.state.FunctionInitializationContext;
+import org.apache.flink.runtime.state.FunctionSnapshotContext;
+import org.apache.flink.streaming.api.checkpoint.CheckpointedFunction;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.flink.streaming.api.operators.StreamingRuntimeContext;
+import org.apache.iotdb.service.rpc.thrift.TSStatus;
+import org.apache.iotdb.session.Session;
+import org.apache.iotdb.tsfile.file.metadata.enums.CompressionType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSEncoding;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class IoTDBSink<IN> extends RichSinkFunction<IN> implements CheckpointedFunction {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger LOG = LoggerFactory.getLogger(IoTDBSink.class);
+
+    private IoTSerializationSchema<IN> serializationSchema;
+    private IoTDBOptions options;
+    private transient Session session;
+
+    private boolean batchFlushOnCheckpoint; // false by default
+    private int batchSize = 100;
+    private List<Event> batchList;
+
+    public IoTDBSink(IoTDBOptions options, IoTSerializationSchema<IN> schema) {
+        this.options = options;
+        this.serializationSchema = schema;
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        batchList = new LinkedList<>();
+
+        session = new Session(options.getHost(), options.getPort(), options.getUser(), options.getPassword());
+        session.open();
+
+        session.setStorageGroup(options.getStorageGroup());
+        for (String sensor : options.getTimeseries()) {
+            session.createTimeseries(sensor, TSDataType.TEXT, TSEncoding.PLAIN, CompressionType.UNCOMPRESSED);
```

Comment:
Hi, is that possible to get the datatype of the sensor?
As IoTDB compresses int/long/float/double better than text, and in some applications, users know what data type their data is.

By the way, it is recommended that using CompressionType.SNAPPY by default.

Reply:
Good idea, I will update later.